sdk/servicebus/Azure.Messaging.ServiceBus/src/Plugins/ServiceBusPlugin.cs

```diff
@@ -29,6 +33,58 @@ public abstract class ServiceBusPlugin
         public virtual ValueTask AfterMessageReceiveAsync(ServiceBusReceivedMessage message) =>
             default;
 
+        internal virtual ValueTask ProcessSendAsync(ServiceBusMessage message, ReadOnlyMemory<ServiceBusPlugin> pipeline)
+        {
+            string pluginType = GetType().Name;
+            try
+            {
+                Logger.PluginCallStarted(pluginType, message.MessageId);
+                BeforeMessageSendAsync(message);
+                Logger.PluginCallCompleted(pluginType, message.MessageId);
+            }
+            catch (Exception ex)
+            {
+                Logger.PluginCallException(pluginType, message.MessageId, ex.ToString());
+                throw;
+            }
+            return ProcessSendNextAsync(message, pipeline);
+        }
+
+        internal static ValueTask ProcessSendNextAsync(ServiceBusMessage message, ReadOnlyMemory<ServiceBusPlugin> pipeline)
+        {
+            if (pipeline.Length > 0)
+            {
+                return pipeline.Span[0].ProcessSendAsync(message, pipeline.Slice(1));
+            }
+            return default;
```

Comment:
nit: It is usually better to make guard clauses the 'inner path' in methods. Meaning, you'd check `pipeline.Length == 0` and return `default` in the if-statement. Then, the 'happy path' is the outer-most statement. In methods more than 1 line of code, it is usually the better arrangement.

Reply:
Interesting, I hadn't heard of this rule of thumb. In any case, I will be closing out this PR since we aren't going to be doing this change now.